tests/dbal/migrator_tool_module_test.php

```diff
@@ -118,6 +118,40 @@ public function test_add()
 			$this->fail($e);
 		}
 		$this->assertEquals(true, $this->tool->exists('acp', 'ACP_NEW_CAT', 'ACP_NEW_MODULE'));
+
+		// Test adding module when plural parent module_langname exists
+		// PHPBB3-14703
+		// Adding fail
+		try
+		{
+			$this->tool->add('acp', 'ACP_FORUM_BASED_PERMISSIONS', array(
+				'module_basename'	=> 'acp_new_permissions_module',
+				'module_langname'	=> 'ACP_NEW_PERMISSIONS_MODULE',
+				'module_mode'		=> 'test',
+				'module_auth'		=> '',
+			));
+			$this->fail('Exception not thrown');
+		}
+		catch (Exception $e) {}
```

Comment:
Please do check the actual contents of the exception (the message for example) though